File: python/연속 수열 부분 합 개수.py
def solution(elements):
    lengths = len(elements)
    ans = set()
    
    for i in range(lengths):
        tmp = elements[i]
        ans.add(tmp)
        for j in range(i+1, i+lengths):
            tmp += elements[j%lengths]
            ans.add(tmp)
    return len(ans)

File: js/연속 수열 부분 합 개수.js
function solution(elements) {
    var lengths = elements.length;
    var set = new Set();
    
    for (i=0; i < lengths; i++) {
        var tmp = elements[i];
        set.add(tmp);
        
        for (j=i+1; j < i+lengths; j++) {
            tmp += elements[j%lengths];
            set.add(tmp);
        }
    }
    return set.size;
}